File: smart-form-backend/src/models/Submission.js
const mongoose = require('mongoose');

const submissionSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  formId: {
    type: String,
    required: true,
  },
  formTitle: String,
  data: {
    type: Map,
    of: mongoose.Schema.Types.Mixed,
  },
  documents: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Document',
  }],
  pdfUrl: String,
  status: {
    type: String,
    enum: ['draft', 'submitted', 'completed'],
    default: 'submitted',
  },
}, {
  timestamps: true,
});

submissionSchema.index({ userId: 1, formId: 1 });

module.exports = mongoose.model('Submission', submissionSchema);

File: smart_form_fill_debug/lib/core/models/submission.dart
class Submission {
  final String id;
  final String formId;
  final String formTitle;
  final Map<String, dynamic> data;
  final String? pdfUrl;
  final String status;
  final DateTime createdAt;

  Submission({
    required this.id,
    required this.formId,
    required this.formTitle,
    required this.data,
    this.pdfUrl,
    required this.status,
    required this.createdAt,
  });

  factory Submission.fromJson(Map<String, dynamic> json) {
    return Submission(
      id: json['_id'] ?? '',
      formId: json['formId'] ?? '',
      formTitle: json['formTitle'] ?? '',
      data: Map<String, dynamic>.from(json['data'] ?? {}),
      pdfUrl: json['pdfUrl'],
      status: json['status'] ?? 'submitted',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'formId': formId,
      'formTitle': formTitle,
      'data': data,
      'pdfUrl': pdfUrl,
      'status': status,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
